torch/_inductor/fx_passes/micro_pipeline_tp.py

```diff
@@ -374,23 +374,46 @@ def __post_init__(self):
 
     @classmethod
     def from_match(cls, match: list[torch.fx.Node]) -> "_ScaledMatmul":
+        """
+        Accepts list of torch.fx.Nodes and returns a _ScaledMatmul op.
+        Handles 2 cases:
+
+        1. Single node _scaled_mm op.
+        2. 3 node "reshape -> _scaled_mm -> reshape" pattern.
+        """
         assert len(match) in (1, 3)
         assert match[0].target in (
             aten._scaled_mm.default,
             aten.reshape.default,
         )
-        mm_node = match[0] if len(match) == 1 else match[1]
 
         def get_arg(node: torch.fx.Node, idx: int, default: Any) -> Any:
             if idx >= len(node.args):
                 return default
             return node.args[idx]
 
+        is_reshape_mm_reshape_pattern = match[0].target == aten.reshape.default
+        mm_node = match[1] if is_reshape_mm_reshape_pattern else match[0]
+
+        # case 1: single node match (mm):
```

Comment:
just curious, do we know who is using case 1 and how are they handling incoming shapes of rank 3+?